AStyle addition to travis

0xc0170 opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/latest/reference/guidelines.html#workflow (Pull request template)
-->

This is a new feature - be on master to check the code and be part of Mbed OS 5.9 when most of our codebase actually is conforming to the style. 

Checking only files that has changed within the pull request. More PR will come later to start updating the code base

### Pull request type

<!-- 
    Required
    Please tick only one of the following types. Do not tick more or change the layout.

    [X] Fix
    [ ] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change
-->

[ ] Fix  
[ ] Refactor  
[ ] New target  
[X] Feature  
[ ] Breaking change

0xc0170 commented:
> travis-ci/astyle — Passed, 2 warnings

See travis status, info about how many files changed within this pull request are not conforming to our code style. I added 2 files changed for showcase, will revert once this is approved. 
AStyle only checks files changed and suggests them to be corrected.

With this PR , we should be able to add this check and start checking the incoming files for code style. And starting updating our user facing API to conform (few more PR will come)

0xc0170 commented on drivers/AnalogIn.cpp:
```diff
@@ -25,3 +25,6 @@ SingletonPtr<PlatformMutex> AnalogIn::_mutex;
 };
 
 #endif
+
```
These changes are only to showcase that the script is running properly - only changed files are being checked.

bcostm commented:
> Checking only files that has changed within the pull request.

Good idea but we need a way to skip folders and files from this check. Typically for ST, all the HAL driver files that we copy/paste from ST Cube packages. And the coding style in those files is different from the one in mbed...

0xc0170 commented:
@bcostm The ignore folders are defined in https://github.com/ARMmbed/mbed-os/pull/6590/files#diff-4ed9365cd7635d05b5f90eab986b7a93 (astyleignore file). At the moment targets folder is excluded from the check thus wont affect any HAL/drivers.

geky commented on .travis.yml:
```diff
@@ -94,6 +94,42 @@ matrix:
+      - astyle --version
+      script:
+        # only changed files to be checked for astyle
+        - git diff --name-only $TRAVIS_BRANCH | grep '.*\.\(h\|c\|hpp\|cpp\)' | fgrep -v -f .astyleignore | xargs -n 100 -I {} bash -c "astyle -n --options=.astylerc \"{}\"" > astyle.out;
```
So master will always say "Passed, 0 warnings".

We could also run the astyle branch on the current commit without printing the warnings so the status says "Passed, 204 warnings" on master. On the PR we can make it say "Passed, 204 warnings (+3 warnings)".

0xc0170 replied:
Is that necessary ? Master files are going to be cleaned up all of them (once this is approved, I'll start module by module). Thus master should be always 0. Only PR that modifies the files can be >0. Or?

geky replied:
The problem is this code will say "Passed, 0 warnings" even if there are 100 warnings.

> More PR will come later to start updating the code base

So we don't have 0 warnings yet. Am I wrong?

If we merge this as is your future PRs aren't going to be very impressive. What will they say? "I removed 50 astyle warnings, reducing the total from 0 warnings to 0 warnings"? : )

0xc0170 replied:
Still not completely certain about the solution but would it be:

1. we do astyle on master branch
2. we do astyle on changes (as it is now here)
3. we show these : master + PR astyle warnings (+/- only PR astyle warnings)

Is that correct?

geky replied:
Oh! There status script will grab whatever number you have on master and automatically compare with that.


The only thing that needs to change is that CURR is set to the the total number of warnings.

So you probably just need to run astyle again without the changes.

Note we don't need to print out all of the warnings, the diff is more useful.

geky commented on .travis.yml:
```diff
@@ -94,6 +94,42 @@ matrix:
+          if [ "$PREV" -ne 0 ]
+          then
+              STATUSM="$STATUSM ($(python -c "print '%+d' % ($CURR-$PREV)") warnings)"
+          fi
```
As is, once master has an astyle status, PRs will say "Passed, 4 warnings (+4 warnings)". We should probably remove this script from here if we don't want to compare with master.

cmonr replied:
Why would we not want to compare with master? I would think that having warning delta would be useful, and part of this PR's intent.

cmonr commented on .astylerc:
```diff
@@ -0,0 +1,34 @@
+# Mbed OS code style definition file for astyle
```
Asking for clarification. Some of these flags and comments sound like they modify the files themselves. Is this actually the case?

0xc0170 replied:
modify the files? can you elaborate?

cmonr replied:
Do flags such as `convert-tabs` and `indent-cases` modify files?

geky commented:
![image](https://user-images.githubusercontent.com/1075160/38809986-14799bfe-418e-11e8-9db3-48532b6301ce.png)
@0xc0170!

0xc0170 commented:
Looks good to me ! +3 warnings will be eliminated once I remove those changes (should be done prior CI stage).

0xc0170 commented:
Rebased (latest master) and removed the changes that were testing the script (visible output above in the screen shot https://github.com/ARMmbed/mbed-os/pull/6590#issuecomment-381588356).

time to test this one

0xc0170 commented:
/morph build
